tools/docker/builder/openwrt/Dockerfile

```diff
@@ -0,0 +1,100 @@
+ARG HOME=/home/openwrt
+
+####
+## OpenWrt pre-requisites and "openwrt" user
+####
+FROM ubuntu:18.04 as openwrt-prerequisites
+
+RUN apt-get update \
+    && apt-get -y install --no-install-recommends \
+    build-essential libncurses5 libncurses5-dev python python3 \
+    unzip git ca-certificates gawk wget file bash \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN useradd -ms /bin/bash openwrt
+
+####
+## Pre-built OpenWrt (and extracted sdk)
+####
+FROM openwrt-prerequisites as openwrt-prebuilt
+
+ARG HOME
+COPY --chown=openwrt:openwrt configs/sdk_config $HOME/sdk_config
+USER openwrt
+
+ARG OPENWRT_REPOSITORY=https://git.prpl.dev/prplwrt/prplwrt.git
+ARG OPENWRT_VERSION=9d2efd
+ARG TARGET=mvebu
+
+WORKDIR $HOME
+
+RUN git clone "$OPENWRT_REPOSITORY" openwrt \
+    && cd openwrt \
+    && git checkout "$OPENWRT_VERSION"
+
+WORKDIR $HOME/openwrt
+
+RUN cp "${HOME}/sdk_config" .config \
+    && echo "CONFIG_TARGET_${TARGET}=y" >> .config \
+    && cat .config \
+    && make defconfig \
+    && make -j$(nproc) 2>&1 | tee build.log
+
+RUN TAR="$(find bin/targets -name 'openwrt-sdk*' -exec echo {} \;)" \
+    && mkdir -p "${HOME}/openwrt_sdk" \
+    && mv -v "$TAR" "${HOME}/openwrt_sdk" \
+    && TAR="$(basename $TAR)" \
+    && cd "${HOME}/openwrt_sdk" \
+    && tar -xf "$TAR" --strip-components=1 \
+    && rm "$TAR"
+
+####
+## Prebuilt sdk only
+####
+FROM openwrt-prerequisites as openwrt-sdk
+ARG HOME
+COPY --from=openwrt-prebuilt --chown=openwrt:openwrt $HOME/openwrt_sdk $HOME/openwrt_sdk
+WORKDIR $HOME/openwrt_sdk
+
+####
+## Prebuilt sdk and prplmesh dependencies (and some debug tools)
+####
+
+FROM openwrt-sdk as prplmesh-builder
+
+ARG HOME
+COPY --chown=openwrt:openwrt configs/prplmesh_config $HOME/prplmesh_config
+ARG PRPLMESH_VERSION=82f1e83
+ARG TARGET_PLATFORM_TYPE=turris-omnia
+ARG PRPL_FEED=https://git.prpl.dev/prplmesh/feed-prpl.git
+# Depending on the target, you may want to use dwpal instead of nl80211:
+ARG USE_DWPAL=false
+
+RUN apt-get update \
+    && apt-get -y install --no-install-recommends \
+    python3 python-yaml rsync less vim libgetopt-complete-perl \
+    && rm -rf /var/lib/apt/lists/*
+    # Building tlvfs currently uses the docker
+    # container's python (not the one from OpenWrt)
```

Comment:
I would put all this in the base image already. Although I wonder why less, vim and libgetopt-complete are useful..

Reply:
If we ever need to add a new build-time dependency (one that could reasonably taken from the host) or add other debugging tools, we could do so without recompiling the base image (the `openwrt-prebuilt` one takes significantly more time to build than the other ones).

`less`, `vim` et al. have been added to be able to debug failing prplMesh builds (when running the container of course, not when buiding) interactively. 
I don't remember why `libgetopt-complete` though. I'll have to check if it's still used and either remove it or add a comment.

On the other hand it's not like we can keep the same base image forever, we will have to update it from time to time anyway.

What do you think? Is it worth keeping these packages in a separate step?

Reply:
> If we ever need to add a new build-time dependency (one that could reasonably taken from the host) or add other debugging tools, we could do so without recompiling the base image (the `openwrt-prebuilt` one takes significantly more time to build than the other ones).

We can still use this approach when we need that. But I think even then we'll at some point consolidate it into the base image.

> `less`, `vim` et al. have been added to be able to debug failing prplMesh builds (when running the container of course, not when buiding) interactively.

OK, good reason to keep them. Add a comment about it though.

Reply:
Alright, I moved them to the base image.
NB: `libgetopt-complete-perl` was not needed anymore.